server/api/ens/avatar.get.ts

```diff
@@ -0,0 +1,66 @@
+import { z } from 'zod';
+import { CACHE_TTL } from '~/server/utils/cache';
+import { createImageCacheKey } from '~/server/utils/cache-keys';
+import { handleApiError } from '~/server/utils/errors';
+import { handleConditionalRequest, invalidateImageCache, streamImageWithCache } from '~/server/utils/image-cache';
+import { deduplicatedFetch } from '~/server/utils/request-dedup';
+import { useLogger } from '~/utils/use-logger';
+
+const logger = useLogger('ens-avatar-proxy');
+
+// Request validation schema
+const querySchema = z.object({
+  _invalidate: z.string().optional(), // Force cache invalidation
+  name: z.string().regex(/^[\dA-Za-z-]+\.eth$/, 'Invalid ENS name format'),
```

Comment:
hmm would this validation allow `.eth` ens names?

Reply:
you mean non .eth ens name? let me adjust it